### DISCUSSION TITLE
How do I properly turn milvus off ? How to I restore milvus after an unexpected power out?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#18369] [Q&A and General discussion] [answered]
How do I properly turn milvus off ?
Is docker-compose down enough?
Or is it better to first unload all schemas? 

Is there a way to restore milvus after an unexpected power out? 
My computer suddenly turned off, and I had to delete volumes folder to get it working again.
[log.txt](https://github.com/milvus-io/milvus/files/9165516/log.txt)

**Comment by yhmo** [comment, marked as answer]
If you deploy `milvus by docker-compose up -d`, then `docker-compose down` can stop the milvus server. 
Helm/k8s is recommended to deploy milvus cluster. Is any node is down,  k8s can automatically restart it. After restart, milvus node(querynode,datanode,indexnode) can restore itself and back to service.

Your mivus version is 2.0.2, from the log, it seems milvus failed to connect etcd service with timeout, something wrong with the etcd service:
```
milvus-standalone  | [2022/07/22 07:07:37.678 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.678 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | [2022/07/22 07:07:37.679 +00:00] [DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
milvus-standalone  | {"level":"warn","ts":"2022-07-22T07:07:42.682Z","logger":"etcd-client","caller":"v3@v3.5.0/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000531500/#initially=[etcd:2379]","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp: lookup etcd: Temporary failure in name resolution\""}
```
The error "Temporary failure in name resolution" indicates the address cannot be resolved, you can search this message by Google for more details.
